NYTimesMostViewedArticles/NetworkClient.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  NetworkClient.swift
+//  NYTimesMostViewedArticles
+//
+//  Created by Sagar Pahwa on 17/11/20.
+//  Copyright © 2020 Sagar Pahwa. All rights reserved.
+//
+
+import Foundation
+
+protocol NetworkClientType {
+    func getMostViewedArticles(period: PeriodSection, completion: @escaping (Response?, Error?)->())
+    
+    func url(period: PeriodSection) -> URL
+}
+
+enum Keys: String {
+    case nyTimesKey = "yxPlqAEPWW3qKi0BgofuzzXeEbxkGfj4"
+    case mockKey = "1234567890"
+}
+
+enum PeriodSection: Int {
+    case day = 1
+    case week = 7
+    case month = 30
+}
+
+struct API {
+    static let baseURL = "http://api.nytimes.com/"
+    static let contentURL = "svc/mostpopular/v2/viewed"
+        //"svc/mostpopular/v2/mostviewed/all-sections/"
+}
+
+
+typealias closure = ()
+
+extension NetworkClientType {
+    func url(period: PeriodSection) -> URL {
+        let string = "\(API.baseURL)/\(API.contentURL)/\(period.rawValue).json?api-key=\(Keys.nyTimesKey.rawValue)"
+        return URL(string: string)!
+    }
+}
+
+class NetworkClient: NetworkClientType {
```

Comment:
Currrent name `NetwokClient` is a verify generic name. Please use descriptive and contextual name here. As using a generic name will signify that all API calls being made from here this class will be heavily overloaded.

Reply:
Yes it will be used for all api calls that why i have used generics in its implementation, in case of adding more variations of interaction with api like multi part we can refactor it, we have already services dedicated to specific apis, let me know if i misunderstood anything.